server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/DateHistogramAggregator.java

```diff
@@ -201,6 +217,55 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
         };
     }
 
+    @Override
+    public StarTreeBucketCollector getStarTreeBucketCollector(LeafReaderContext ctx, CompositeIndexFieldInfo starTree) throws IOException {
+
+        return new StarTreeBucketCollector() {
+
+            public void setSubCollectors() throws IOException {
+                for (Aggregator aggregator : subAggregators) {
+                    this.subCollectors.add(((StarTreePreComputeCollector) aggregator).getStarTreeBucketCollector(ctx, starTree));
```

Comment:
What happens if sub collectors are not of type `StarTreePreComputeCollector` ? 
will we validate and throw an exception and probably return false in `preCompute` method ?

Reply:
Basically we will not set star-tree query context in the first place itself when the query shape is not supported - so preCompute will not get executed in that case.

Reply:
Agreed -- it feels like there's an `instanceof` check somewhere. 

(I was going to ask what value the interface provides if you never cast anything to it, because it wasn't clear where it was being used.)

Reply:
Yeah, the checks are actually at AggregatorFactory level for now which recognize if a given query shape is resolvable via star-tree or not. Can think of a better design to identify resolvable query shapes as a follow-up.